src/Blockcore.Indexer.Core/Storage/Mongo/BlockRewindOperation.cs

```diff
@@ -0,0 +1,187 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Blockcore.Indexer.Core.Storage.Mongo.Types;
+using MongoDB.Bson;
+using MongoDB.Driver;
+
+namespace Blockcore.Indexer.Core.Storage.Mongo;
+
+public static class BlockRewindOperation
+{
+   public static async Task RewindBlockOnIbdAsync(this MongoData storage, long blockIndex)
+   {
```

Comment:
this method seems complicated, I know its an edge case but later we defo need tests here.